[ADD] Plan Definition

rgarnau opened the pull request:
This module adds the concept of Plan Definition (https://www.hl7.org/fhir/plandefinition.html).

A Plan Definition is a pre-defined group of actions to be taken in particular circumstances. Each Action describes an activity to be performed (in terms of an Activity Definition resource).

@lasley @obulkin @jbeficent

This module is still 'Work in progress' as tests are missing.

rgarnau committed:
[ADD] Plan Definition

rgarnau committed:
[FIX] little fix

obulkin commented:
As with the other HL7 PRs, I'm happy to review more closely once this is finished, but in the meantime, here are some high-level thoughts:

- Why do activity definitions have two different ``product.product`` fields and a ``product.category`` one? At most, it seems like activities should be connected to one product, with a related field to surface the category of that product, if necessary.
- The direct connection between a plan and a single activity (``activity_definition_id``) doesn't match up with HL7, where activity definitions seem to act as templates for actions and are only connected to plan definitions through the plan's actions.

rgarnau committed:
[FIX] removed product_id and categ_id from activity definition

rgarnau commented:
Hi @obulkin, thanks for the review!
- You are right, one of the two product.product is redundant (I have removed it). The product.category is something we implemented more specifically for personal needs (we carefully reviewed the FHIR standard but there wasn't a proper solution to this) but this is better implemented in that particular module.
- That's also true. According to FHIR, a Plan Definition defines a list of actions each of them corresponding to either Request Groups or Procedure Requests. However, for the sake of simplicity, in our implementation we managed to assign an Activity Definition to a Plan Definition so it would correspond to the Request Group itself and all the actions inside would correspond to Procedure Requests. Obviously, in our case, we just considered one level of actions.

I've created a new branch with the Plan Definition following more strictly the FHIR standard as you mention in #194. I'm also closing this PR, since it is not useful anymore.

@jbeficent

obulkin commented:
Thanks, @rgarnau. I'll take a look at the new PR.
